src/model/vision_fusion.py

```
@@ -0,0 +1,460 @@
+
+    Inserts vision features at specified positions in text sequence.
+
+    Args:
+        vision_features: Vision tokens (batch, num_vis, dim)
+        text_embeddings: Text embeddings (batch, num_text, dim)
+        vision_positions: Positions to insert vision (list of indices)
+
+    Returns:
+        Interleaved sequence
+    """
+    batch_size = text_embeddings.shape[0]
```

Ruff (0.14.2)

Local variable `batch_size` is assigned to but never used

Remove assignment to unused variable `batch_size`

(F841)